### DISCUSSION TITLE
Recursive component

### DISCUSSION THREAD

**Discussion by chosten** [alpinejs/alpine#751] [4. General]
Hello,
I can't seem to find a way to create recursive components. Is it possible with alpine ?
Something like a tree component for example. I can render one level no problem but I'm stuck trying to render the nodes recursively.

**Comment by SimoTod** [comment]
Alpine doesn't support it. You need to find a third party library and instantiate it in an x-init directive

**Comment by mhussa** [comment]
see [this](https://codepen.io/ScottWindon/pen/RwabppB)

**Reply by leopku** [reply]
Thank you @mhussa for the amazing example.

But could you please look at https://codepen.io/leopku/pen/MWjVEyw ?
This example had little changes which load AJAX data in `x-init` and It broken.

Could you please help me to find the point it broken?

**Reply by KevinBatdorf** [reply]
@leopku You need to use an arrow function to keep the scope of this to the component. Update your code to this:

```js
.then(data => {
  // console.log(data.json);
  this.levels = data.json;
})
```

**Reply by leopku** [reply]
@KevinBatdorf Thanks a million.

**Reply by mhussa** [reply]
> Thank you @mhussa for the amazing example.

To be clear, the example was written and shared by this righteous dude:  https://codepen.io/ScottWindon

**Reply by leopku** [reply]
> > Thank you @mhussa for the amazing example.
> 
> To be clear, the example was written and shared by this righteous dude: https://codepen.io/ScottWindon

My mistake. Thanks Scott Windon.

**Comment by michaelsboost** [comment]
For anyone else having issues with this. I made a Codepen showing how I was able to do it.

https://codepen.io/michaelsboost/pen/bGPdNMX
